### DISCUSSION TITLE
Prefer new issues to expanding on the old ones?

### DISCUSSION THREAD

**Discussion by mnalis** [streetcomplete/StreetComplete#3011] [Repository Maintenance] [answered]
in https://github.com/streetcomplete/StreetComplete/issues/1961#issuecomment-871396759, @matkoniecz wrote:

> 
> @mnalis If there are some specific improvements to be made it would be better to open a new issue for them. Long comments threads in already closed issues are (at least for me) not the best communication method, as it is possible to discuss endlessly. With multiple, amorphous, unclear or even never defined problem.
> 
> If there is some specific problem or enhancement - maybe open a new issue for that?
> 
> I get notifications about any comment made on StreetComplete issue tracker and in such cases I need to reread entire discussion to even know what is being discussed.
> 
> Maybe for others this would be preferable to discuss new related problems in the initial issue but I will likely start disabling notifications for discussions in old closed issues as it is far too time consuming to follow them.
> 
> To be clear: (1) maybe it is just my own preference? (2) I am not saying that such feedback is not valuable - just that I am not capable of processing ongoing commentary in old issues and it seems to me that opening new issues for that would be better (if there is no specific issue for that already)

**Comment by mnalis** [comment]
@matkoniecz oh, I understand. I will be opening new issues directly in the future, if that is preferred.

I copied this meta-discussion manually (as I don't know if there is better way) from https://github.com/streetcomplete/StreetComplete/issues/1961#issuecomment-871396759, as it seems to be to be even more meta-related, and not really and issue about SC quest so probably this github "Discussions" is better fit?

> If there is some specific problem or enhancement - maybe open a new issue for that?

I was just about to fork that `ask about cycleway on lanes=1 crossing:island=yes`  from https://github.com/streetcomplete/StreetComplete/issues/1961#issuecomment-871343532 into new issue - if people confirmed that would be useful, that is why I said:

> I'd open new quest if it seems ok

As for the original `pedestrian island is too spammy` comment from https://github.com/streetcomplete/StreetComplete/issues/1961#issuecomment-869216508, I was actually trying to be more helpful / less spammy this way commenting on old ticket, my rationale being:

- it seemed easier to me to track one discussion in one issue, then trying to correlate chronologically several separate issues in several different browser tabs that talk about same thing (eg. this issue even initially mentioned `Not an overwhelming percentage of elements have the same answer (No spam): I'm not sure about this one. I think most crossings won't have a pedestrian island, but quite a few do. That might vary by region though.` and I was going to expand on that. (`it is quite spammy in Croatia`)
- also in future only one issue for that needs to be found/referenced, instead of several of them
- it looked more targeted and thus less spammy to people - people who created/subscribed to this issue were probably interested in it, while those who were not subscribed (or manually unsubscribed) would not be interested in it and thus would not be notified. And new issue looked more like shotgun-effect, messaging everybody who tracks new issues, even if they already unsubscribed for that specific thematic several times.

**Comment by mnalis** [comment]
So, What would people (especially devels working on SC, and @westnordost primarily) prefer in such cases?

- like myself tried, to comment on old issue related to source of the problem (quest too spammy), which progressed collecting enough information  to open a new issue for new quest / updated quest `ask about cycleway for lanes=1 with crossing:island=yes`? Not ideal as there are long threads then, but it does generate the smallest cumulative amount on text overall (As there is no need to re-summarize all that was said before) 

- like @matkoniecz, to open new Issue(s) straight out (so in this example, it would firstly be new issue about `pedestrian island is too spammy` and when discussion in that one advances, then new  issue for `ask about cycleway for lanes=1 with crossing:island=yes` ? And then just link old/related issues so interested parties (and later searches) find it linked it together?

- First open the discussion (with link to original issue), and then depending on the feedback open new issues (linking to that discussion)? Probably looks the cleanest (most separated) to me now to address @matkoniecz  concerns, but cumulative amount of text to read would actually probably be biggest.

- establish some threshold about interesting issues, where this example would be below the bar as too uninteresting, and refrain from even reporting such things? Might reduce load the most, but seems like very hard thing to define (besides being brutally honest to someone and saying _"**nothing** you talk about is **ever** interesting enough"_, which I sincerely hope is the level I have not yet reached :smile:)

I'm painfully aware of amount of load the developers get (and, ironically, that even this discussion is adding to that load!), but would really like some "rules of engagement" to be the best possible contributor I can be (without becoming android developer myself).

**Comment by westnordost** [comment, marked as answer]
Uhm... yes I agree. 

An issue should be actionable, everything else is a discussion. So, if there is something concrete to do, like that with the crossing island, then better create a new ticket for that. It is always good of course to link back to related issues for context. Better yet, of course, if you think there is not much of a discussion to be had about the proposed change, as a PR.

If a comment is added to a (closed) issue, I usually ignore it, because this falls under "discussion" in my thinking - so if any conclusion is reached, I expect a new issue or PR to be opened. Otherwise it is just too much (for me to follow).

That having said, there is the yet unspoken hope / expectation that those that are very active in the issue tracker slowly become acquainted with the codebase themselves and someday also become active contributors or maybe at one time co-maintainers.

**Reply by andrewharvey** [reply]
> That having said, there is the yet unspoken hope / expectation that those that are very active in the issue tracker slowly become acquainted with the codebase themselves and someday also become active contributors or maybe at one time co-maintainers.

As someone who's recently been very vocal on the issue tracker and not yet doing any code contributions, ￼￼I'd really like to help out via PR, I'm just waiting on one of my quest suggestions to be accepted, or one of my suggestions for existing tweek changes to loosen their criteria to be accepted. There's no point doing the work coding it up if ultimately it's not accepted by the maintainers per issue tracker discussion. I thought about doing a fork with the policy that any valid tagging quest is accepted but deemed it would be too much effort to try and keep in sync with mainline and ongoing releases.
